src/core/class_db.cpp

```diff
@@ -352,8 +352,8 @@ void ClassDB::initialize(GDExtensionInitializationLevel p_level) {
 }
 
 void ClassDB::deinitialize(GDExtensionInitializationLevel p_level) {
-	for (std::vector<StringName>::reverse_iterator i = class_register_order.rbegin(); i != class_register_order.rend(); ++i) {
-		const StringName &name = *i;
+	std::set<StringName> to_erase;
+	for (const StringName &name : class_register_order) {
```

Comment:
This appears to be looping forward through `class_register_order`. We need to loop _backwards_ through `class_register_order` so that classes are unregistered in the opposite order that they were registered.

Reply:
Ok I thought it was related to the erasure but it's not then. I'll fix it when I find a minute 👍🏽

Reply:
I fixed that loop order, although I kept the order of removal from `class_register_order` itself. Is that fine? Or should I also preserve the order of removal in that container?

Reply:
Thanks! Yep, that should be fine - we only really care about the order the classes are unregistered, it doesn't matter what order they are removed from the `class_register_order` vector (so long as the remaining items are still in the same relative order).